inject/inject/src/main/java/io/helidon/inject/ServiceProviderRegistry.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.inject;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.helidon.inject.service.ServiceInfo;
+
+/**
+ * This is an advanced service registry providing access to service providers, rather than just
+ * {@link java.util.function.Supplier}.
```

Comment:
The javadoc should advise users to attempt to use the other one instead.

Reply:
will fix